sdk/src/calibration_96tof1.cpp

```diff
@@ -205,120 +350,152 @@ aditof::Status Calibration::getIntrinsic(float key,
     return Status::GENERIC_ERROR;
 }
 
-//! SaveCalMap - Save the entire calibration map
+//! setMode - Sets the mode to be used for depth calibration
 /*!
-    SaveCalMap - Saves the entire calibration map as binary to a file.
-    \param calibration_map - Calibarion map to be displayed
-    \param file_name - File where the calibration map needs to be stored
+setMode - Sets the mode to be used for depth calibration
+\param mode - Camera depth mode
+\param range - Max range for selected mode
+\param frameWidth - Width of the depth image in pixels
+\param frameHeight - Height of the depth image in pixels
 */
-aditof::Status
-Calibration::saveCalMap(std::shared_ptr<aditof::DeviceInterface> device) {
+aditof::Status Calibration96Tof1::setMode(const std::string &mode, int range,
+                                          unsigned int frameWidth,
+                                          unsigned int frameheight) {
     using namespace aditof;
 
-    std::vector<float> data;
-    for (const auto &mapElement : m_calibration_map) {
-        data.push_back(mapElement.first);
-        const packet_struct &sub_packet_map = mapElement.second;
-        data.push_back(sub_packet_map.size);
-
-        for (const auto &packet : sub_packet_map.packet) {
-            data.push_back(packet.first);       // write parameter key
-            data.push_back(packet.second.size); // write size of parameter
+    Status status = Status::OK;
+    std::vector<float> cameraMatrix;
+    const int16_t pixelMaxValue = (1 << 12) - 1; // 4095
+    float gain = 1.0, offset = 0.0;
 
-            for (const auto &value : packet.second.value) {
-                data.push_back(
-                    static_cast<float>(value)); // write parameter values
-            }
-        }
+    status = getGainOffset(mode, gain, offset);
+    if (status != Status::OK) {
+        LOG(WARNING) << "Failed to read gain and offset from eeprom";
+        return status;
+    } else {
+        LOG(INFO) << "Camera calibration parameters for mode: " << mode
+                  << " are gain: " << gain << " "
+                  << "offset: " << offset;
     }
+    buildDepthCalibrationCache(gain, offset, pixelMaxValue, range);
+    m_range = range;
 
-    float size = static_cast<float>(data.size() * sizeof(uint32_t));
-    device->writeEeprom((uint32_t)0, (uint8_t *)&size, (size_t)4);
-    device->writeEeprom((uint32_t)4, (uint8_t *)data.data(), size);
+    status = getIntrinsic(INTRINSIC, cameraMatrix);
+    if (status != Status::OK) {
+        LOG(WARNING) << "Failed to read intrinsic from eeprom";
+        return status;
+    } else {
+        LOG(INFO) << "Camera intrinsic parameters:\n"
+                  << "    fx: " << cameraMatrix[0] << "\n"
+                  << "    fy: " << cameraMatrix[4] << "\n"
+                  << "    cx: " << cameraMatrix[2] << "\n"
+                  << "    cy: " << cameraMatrix[5];
+    }
+    buildGeometryCalibrationCache(cameraMatrix, frameWidth, frameheight);
```

Comment:
This could be done only once (when frame types gets changed), not every time the mode is changed.

Reply:
I'd keep it like this for now since the calibration class does not have a function to let it know that the frame type was changed or a public function to trigger the rebuild of the geometry cache.

Reply:
ok